### PR TITLE
Fix DOM memory leak (second attempt)

### PR DESCRIPTION
**Pull request by mveytsman**
This is an alternative approach to https://github.com/phoenixframework/phoenix_live_view/pull/1079 as discussed in https://github.com/phoenixframework/phoenix_live_view/pull/1079#discussion_r456942016

It fixes https://github.com/phoenixframework/phoenix_live_view/issues/1078

### CODE REVIEW

File: assets/js/phoenix_live_view.js [hunk 1]
```diff
@@ -1271,6 +1271,21 @@ export let DOM = {
     } else {
       return true
     }
+  },
+
+  cleanChildNodes(container, phxUpdate){
+    if (DOM.isPhxUpdate(container, phxUpdate, ["append", "prepend"])) {
+      Array.from(container.childNodes).forEach(childNode => {
+        if (childNode.nodeType == Node.TEXT_NODE && childNode.nodeValue.trim() == "") {
```
Comment (josevalim) [inline assets/js/phoenix_live_view.js:1279]: I think we can remove them even if they have text. We just need to warn if they have text too. So all non text nodes and all text nodes with contents are warned. WDYT?

File: assets/js/phoenix_live_view.js [hunk 2]
```diff
@@ -1352,7 +1367,7 @@ class DOMPatch {
         },
         onBeforeNodeDiscarded: (el) => {
           if(el.getAttribute && el.getAttribute(PHX_REMOVE) !== null){ return true }
-          if(el.parentNode !== null && DOM.isPhxUpdate(el.parentNode, phxUpdate, ["append", "prepend"])){ return false }
+          if(el.parentNode !== null && DOM.isPhxUpdate(el.parentNode, phxUpdate, ["append", "prepend"]) && el.id){ return false }
```
Comment (josevalim) [inline assets/js/phoenix_live_view.js:1370]: Theoretically we don't need this change anymore, right?

Comment (mveytsman) [reply assets/js/phoenix_live_view.js:1370]: We do, but the reason is subtle.

If our initial render has a bunch of divs and text nodes, and we append with an update that has just one div, we're going to clean the text nodes from the `toEl` but not from the `fromEl`, so they will still remain there.

The number of text nodes will not grow unbounded without this change, but if we want to have the `phx-update` container to only have children with IDs we need this.

File: assets/js/phoenix_live_view.js (hunk 1 above)
Comment (mveytsman) [reply assets/js/phoenix_live_view.js]: This code did that but it wasn't very clear in the intent. I rewrote it to be clearer.

**Comment by chrismccord** [comment]
❤️❤️❤️🐥🔥

**Pull request merged**